packages/memory-storage/test/no-crash-on-big-buffers.test.ts

```diff
@@ -0,0 +1,39 @@
+// https://github.com/apify/crawlee/issues/1732
+// https://github.com/apify/crawlee/issues/1710
+
+import { rm } from 'node:fs/promises';
+import { resolve } from 'node:path';
+import { MemoryStorage } from '@crawlee/memory-storage';
+import type { KeyValueStoreClient, KeyValueStoreInfo } from '@crawlee/types';
+
+describe('MemoryStorage should not crash when saving a big buffer', () => {
+    const tmpLocation = resolve(__dirname, './tmp/no-buffer-crash');
+    const storage = new MemoryStorage({
+        localDataDirectory: tmpLocation,
+        persistStorage: false,
+    });
+
+    let kvs: KeyValueStoreInfo;
+    let store: KeyValueStoreClient;
+
+    beforeAll(async () => {
+        kvs = await storage.keyValueStores().getOrCreate();
+        store = storage.keyValueStore(kvs.id);
+    });
+
+    afterAll(async () => {
+        await rm(tmpLocation, { force: true, recursive: true });
+    });
+
+    test('should not crash when saving a big buffer', async () => {
+        const numbers = Array.from(([...Array(18_100_000).keys()]).map((i) => i * 3_000_000));
```

Comment:
just curious how long does this test run, otherwise ready to merge

Reply:
Too long sadly. Maybe we should skip it?

![Code - 2023-01-10 at 10 56 37@2x](https://user-images.githubusercontent.com/17960496/211506004-abc59723-8ea9-4ee3-9520-bbe69bf6e807.png)

Reply:
Turns out its too big of a buffer for CI 😢. Should we skip it? Make it an e2e test instead?

Reply:
if you cant find a way to mock this somehow, i'd just skip it, or maybe leave it there but use much smaller file, so the test is fast, and have a comment to enable the slow mode for local testing that actually reproduced the problem